router/core/operation_processor.go

```diff
@@ -822,6 +811,69 @@ func (o *OperationKit) NormalizeVariables() error {
 	return nil
 }
 
+func (o *OperationKit) RemapVariables() error {
+	before := len(o.kit.doc.Input.RawBytes)
+
+	report := &operationreport.Report{}
+	variablesMap := o.kit.variablesRemapper.NormalizeOperation(o.kit.doc, o.operationProcessor.executor.ClientSchema, report)
+	if report.HasErrors() {
+		return &reportError{
+			report: report,
+		}
+	}
+	o.parsedOperation.RemapVariables = variablesMap
+
+	// Hash the normalized operation with the static operation name to avoid different IDs for the same operation
+	err := o.kit.printer.Print(o.kit.doc, o.kit.keyGen)
+	if err != nil {
+		return errors.WithStack(fmt.Errorf("RemapVariables failed generating operation hash: %w", err))
+	}
+
+	// Print the operation without the operation name to get the pure normalized form
+	// Afterward we can calculate the operation ID that is used as a stable identifier for analytics
+
+	o.kit.normalizedOperation.Reset()
+	// store the original name of the operation
+	nameRef := o.kit.doc.OperationDefinitions[o.operationDefinitionRef].Name
+
+	staticNameRef := o.kit.doc.Input.AppendInputBytes([]byte(""))
+	o.kit.doc.OperationDefinitions[o.operationDefinitionRef].Name = staticNameRef
+
+	err = o.kit.printer.Print(o.kit.doc, o.kit.normalizedOperation)
+	if err != nil {
+		return err
+	}
+	// Reset the doc with the original name
+	o.kit.doc.OperationDefinitions[o.operationDefinitionRef].Name = nameRef
+
+	o.kit.keyGen.Reset()
+	_, err = o.kit.keyGen.Write(o.kit.normalizedOperation.Bytes())
+	if err != nil {
+		return err
+	}
+
+	// Generate the operation ID
+	o.parsedOperation.InternalID = o.kit.keyGen.Sum64()
+	o.kit.keyGen.Reset()
+
+	// If the normalized form of the operation didn't change, we don't need to print it again
+	after := len(o.kit.doc.Input.RawBytes)
+	if after == before {
```

Comment:
This checks length, is this robust against same number of characters but different names or different values after normalization? maybe better to hash it?